src/coreclr/inc/profilepriv.inl

```diff
@@ -203,77 +210,105 @@ inline void ProfControlBlock::DeRegisterProfilerInfo(ProfilerInfo *pProfilerInfo
+    return pProfilerInfo->pProfInterface->IsCallback3Supported(); 
 }
 
-inline BOOL ProfControlBlock::IsCallback5Supported()
+FORCEINLINE BOOL ProfControlBlock::IsCallback3Supported()
 {
-    return AnyProfilerPassesCondition([](ProfilerInfo *pProfilerInfo) { return pProfilerInfo->pProfInterface->IsCallback5Supported(); });
+    return AnyProfilerPassesCondition(&IsCallback3SupportedHelper);
 }
 
-inline BOOL ProfControlBlock::IsDisableTransparencySet()
+inline BOOL IsCallback5SupportedHelper(ProfilerInfo *pProfilerInfo)
 {
-    return AnyProfilerPassesCondition([](ProfilerInfo *pProfilerInfo) { return pProfilerInfo->eventMask.IsEventMaskSet(COR_PRF_DISABLE_TRANSPARENCY_CHECKS_UNDER_FULL_TRUST); });
+    return pProfilerInfo->pProfInterface->IsCallback5Supported();
 }
 
-inline BOOL ProfControlBlock::RequiresGenericsContextForEnterLeave()
+FORCEINLINE BOOL ProfControlBlock::IsCallback5Supported()
 {
-    return AnyProfilerPassesCondition([](ProfilerInfo *pProfilerInfo) { return pProfilerInfo->pProfInterface->RequiresGenericsContextForEnterLeave(); });
+    return AnyProfilerPassesCondition(&IsCallback5SupportedHelper);
 }
 
-inline bool DoesProfilerWantEEFunctionIDMapper(ProfilerInfo *pProfilerInfo)
+inline BOOL RequiresGenericsContextForEnterLeaveHelper(ProfilerInfo *pProfilerInfo)
 {
-    CONTRACTL
-    {
-        NOTHROW;
-        GC_NOTRIGGER;
-        CANNOT_TAKE_LOCK;
-    }
-    CONTRACTL_END;
+    return pProfilerInfo->pProfInterface->RequiresGenericsContextForEnterLeave();
+}
+
+FORCEINLINE BOOL ProfControlBlock::RequiresGenericsContextForEnterLeave()
+{
+    return AnyProfilerPassesCondition(&RequiresGenericsContextForEnterLeaveHelper);
+}
 
+inline bool DoesProfilerWantEEFunctionIDMapper(ProfilerInfo *pProfilerInfo)
+{
     return ((pProfilerInfo->pProfInterface->GetFunctionIDMapper()  != NULL) ||
              (pProfilerInfo->pProfInterface->GetFunctionIDMapper2() != NULL));
 }
 
+inline void EEFunctionIDMapperHelper(ProfilerInfo *pProfilerInfo, FunctionID funcId, BOOL *pbHookFunction, UINT_PTR *pPtr)
+{
+   if (DoesProfilerWantEEFunctionIDMapper(pProfilerInfo))
+   {
+       *pPtr = pProfilerInfo->pProfInterface->EEFunctionIDMapper(funcId, pbHookFunction);
+   }
+}
+
 inline UINT_PTR ProfControlBlock::EEFunctionIDMapper(FunctionID funcId, BOOL *pbHookFunction)
 {
     LIMITED_METHOD_CONTRACT;
     UINT_PTR ptr = NULL;
     DoOneProfilerIteration(&mainProfilerInfo,
                           ProfilerCallbackType::Active,
-                          [](ProfilerInfo *pProfilerInfo, FunctionID funcId, BOOL *pbHookFunction, UINT_PTR *pPtr)
-                          {
-                               if (DoesProfilerWantEEFunctionIDMapper(pProfilerInfo))
-                               {
-                                   *pPtr = pProfilerInfo->pProfInterface->EEFunctionIDMapper(funcId, pbHookFunction);
-                               }
-                          },
+                          &EEFunctionIDMapperHelper,
```

Comment:
Is there any observable codegen difference from converting the lambdas to regular functions?